src/store.ts

```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import type { State } from './App';
+
+const Store = React.createContext<State | null>(null);
+export default Store;
```

Comment:
```suggestion
import React from 'react';
import type { State } from './App';

const Store = React.createContext<State | null>(null);
export default Store;


export const StoreConsumer = ({ children }: {children: (state: State) => JSX.Element}) => {
  return (
    <Store.Consumer>
      {(state) => {
        if (state == null) {
          throw new Error(`state is null: ${state}`)
        }

        return children(state);
      }}
    </Store.Consumer>
  )
}
```

요런 StoreConsumer 컴포넌트를 만든다면 state가 nullable한 상황을 피할 수 있지 않을까요?

Reply:
이런 방법이 있었네요! 비록 함수형으로 변경하면서 useContext를 사용해서 해당 함수를 사용하지는 않았지만 알아갑니다~!